puppet-syntax.gemspec

```diff
@@ -20,6 +20,12 @@ Gem::Specification.new do |spec|
 
   spec.add_dependency 'puppet', '>= 7', '< 9'
   spec.add_dependency 'rake', '~> 13.1'
+  # dependency for Puppet on Ruby 3.4
+  spec.add_dependency 'base64', '~> 0.2'
+  spec.add_dependency 'getoptlong', '~> 0.2'
+  spec.add_dependency 'syslog', '~> 0.3'
+  # dependency for Puppet on Ruby 3.3
+  spec.add_dependency 'racc', '~> 1.8', '>= 1.8.1'
```

Comment:
We shouldn't add Puppet's dependencies as runtime dependencies. Please add them to `Gemfile`.